source/core/slang-string.cpp

```diff
@@ -45,6 +45,23 @@ namespace Slang
 
     // UnownedStringSlice
 
+    bool UnownedStringSlice::startsWith(UnownedStringSlice const& other) const
+    {
+        UInt thisSize = size();
+        UInt otherSize = other.size();
+
+        if (otherSize > thisSize)
+            return false;
+
+        return UnownedStringSlice(begin(), begin() + otherSize) == other;
```

Comment:
I suppose that works, since `OwnedStringSlice` could still use the raw pointers from `UnownedStringSlice` and just add the retained `String` to that, so long as we are sure that the actual underlying allocation for the `String` can't move while we have a retained pointer to it (which I suppose *is* guaranteed).